replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/ReplicatorRunner.java

```diff
@@ -46,6 +50,8 @@
 
   private final Replicator replicator;
 
+  private final Heartbeater heartbeater;
```

Comment:
Based on my discussions with Chris, this seems like the central point. Originally, we were thinking of doing in the same loop but after more discussions with Peter, we moved it to a different thread. The fact is that the verification portion is part of both. And since this is where we dealt with configurations, it seemed appropriate to deal with remote sites/configs from here while leaving the actual logic in the heartbeater